lib/travis/model/job/queue.rb

```diff
@@ -8,119 +8,81 @@ class Job
   # queue) or the language given in the configuration (`.travis.yml`) and
   # default to 'builds.linux'.
   class Queue
+    SUDO_AND_SETUID_EXECUTABLES = %w(
+      sudo
+      ping
+    )
+
+    SUDO_DETECTION_REGEXP = /\A[^#]*\b(#{SUDO_AND_SETUID_EXECUTABLES.join('|')})\b/
```

Comment:
Thinking about this some more, I wonder if we want `^` instead of `\A`. For example, this string wouldn't match, but I think it should:

```
# this is something that we need sudo for
sudo echo test
```